### ISSUE TITLE
Handle MechEngineer explosion damage

### ISSUE THREAD

**Issue by Sheep-y** [Sheep-y/Attack-Improvement-Mod#22]
```
MechEngineer [ERROR] System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NullReferenceException: Object reference not set to an instance of an object
at Sheepy.BattleTechMod.AttackImprovementMod.AttackLog.GetHitKey (string,object,string) <0x00015>
at Sheepy.BattleTechMod.AttackImprovementMod.AttackLog.LogMechDamage (BattleTech.Mech,BattleTech.ArmorLocation,BattleTech.Weapon) <0x0012d>
at (wrapper dynamic-method) BattleTech.Mech.DamageLocation_Patch7 (object,int,BattleTech.WeaponHitInfo,BattleTech.ArmorLocation,BattleTech.Weapon,single,int,BattleTech.AttackImpactQuality,BattleTech.DamageType) <0x02072>
at (wrapper dynamic-method) BattleTech.Mech.DamageLocation_Patch7 (object,int,BattleTech.WeaponHitInfo,BattleTech.ArmorLocation,BattleTech.Weapon,single,int,BattleTech.AttackImpactQuality,BattleTech.DamageType) <0x01ff9>
at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke (object,object[],System.Exception&) <0x00080>
at System.Reflection.MonoMethod.Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo) <0x001ac>

  --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <filename unknown>:0 
  at Harmony.Traverse.GetValue () [0x00000] in <filename unknown>:0 
  at MechEngineer.ComponentExplosionHandler.CheckForExplosion (BattleTech.MechComponent component, WeaponHitInfo hitInfo, ComponentDamageLevel damageLevel, Boolean applyEffects) [0x00000] in <filename unknown>:0 
  at MechEngineer.MechComponent_DamageComponent_Patch2.Postfix (BattleTech.MechComponent __instance, WeaponHitInfo hitInfo, ComponentDamageLevel damageLevel, Boolean applyEffects) [0x00000] in <filename unknown>:0 
```

**Comment by Sheep-y** [comment]
WeaponRealizer has more than heat to damage, and a pull request has been made so that AIM can call it to get prediction:

https://github.com/janxious/BT-WeaponRealizer/pull/5

**Comment by Sheep-y** [comment]
MechEngineer ammo explosion (and other non-standard combat damages) filtered in commit 2818105e9b5f9bbd44e23011840dc0d9cfc4d6f8.

**Comment by Sheep-y** [comment]
WeaponRealizer bridge implemented in commit 5ea0446c07c72787952558d6c800e6ac19d805f3.
